### ISSUE TITLE
Custom loss function for keras (or mlp)

### ISSUE THREAD

**Issue by daniel-althoff** [tidymodels/parsnip#777]
## Feature

(1) When training a mlp() (with 'keras' engine), it would be great if we could provide a custom loss function, similar way as we set metric.
This would open up the opportunity for different customizations or event uncertainty quantification with different objective functions.

(2) I saw a post about introducing keras callbacks (https://github.com/tidymodels/parsnip/issues/70#issuecomment-426459570), as early_stopping, I couldn't figure out how to do this. It just throws an error for me that the "others" is a unused argument.

Thank you!
